src/test/java/com/google/cloud/tools/gradle/appengine/core/DownloadCloudSdkTaskTest.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2016 Google Inc. All Right Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.google.cloud.tools.gradle.appengine.core;
+
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.File;
+import java.io.IOException;
+import org.gradle.api.GradleException;
+import org.gradle.api.Project;
+import org.gradle.testfixtures.ProjectBuilder;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DownloadCloudSdkTaskTest {
+
+  @Mock private ToolsExtension toolsExtension;
+  @Mock private CloudSdkBuilderFactory cloudSdkBuilderFactory;
+  @Mock private SdkDownloader sdkDownloader;
+
+  @Rule public ExpectedException exception = ExpectedException.none();
+  @Rule public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  private DownloadCloudSdkTask downloadCloudSdkTask;
+
+  /** Setup DownloadCloudSdkTaskTest. */
+  @Before
+  public void setup() {
+    Project tempProject = ProjectBuilder.builder().build();
+    downloadCloudSdkTask =
+        tempProject.getTasks().create("tempDownloadTask", DownloadCloudSdkTask.class);
+    downloadCloudSdkTask.setCloudSdkBuilderFactor(cloudSdkBuilderFactory);
+    downloadCloudSdkTask.setToolsExtension(toolsExtension);
+    downloadCloudSdkTask.setSdkDownloader(sdkDownloader);
+  }
+
+  @Test
+  public void testDownloadCloudSdkAction_validateTrue() {
+    String version = "LATEST";
+    File home = getTempHomeDirectory("LATEST");
+    when(toolsExtension.getCloudSdkVersion()).thenReturn(version);
+    when(toolsExtension.getCloudSdkHome()).thenReturn(home);
+
+    // Assume SDK installation is valid
+    when(sdkDownloader.sdkIsValid(version, home)).thenReturn(true);
+
+    downloadCloudSdkTask.downloadCloudSdkAction();
+    verify(sdkDownloader, never()).downloadSdk(version);
+    verify(sdkDownloader, times(1)).sdkIsValid(version, home);
```

Comment:
nit: Generally, only methods that have side effects are necessary to verify their calls.